src/main/groovy/au/org/ala/spatial/process/GeneratePoints.groovy

```diff
@@ -74,38 +75,48 @@ class GeneratePoints extends SlaveProcess {
                 new NameValuePair("alaId", userId.toString())
         ]
 
+        task.history.put(System.currentTimeMillis(), "Uploading points to sandbox: ${sandboxBiocacheServiceUrl}")
+
         def response = Util.urlResponse("POST", "${sandboxBiocacheServiceUrl}/upload/",
                 nameValuePairs.toArray(new NameValuePair[0]))
 
         if (response) {
+            if (response.statusCode != 200) {
+                task.message = "Error"
+                task.history.put(System.currentTimeMillis(), response.statusCode + " : " + response.text)
+                return
+            }
             def dataResourceUid = JSON.parse(response.text).uid
-
+            task.history.put(System.currentTimeMillis(), "Sandbox data resource uid:" + dataResourceUid)
             //wait
             def statusUrl = "${sandboxBiocacheServiceUrl}/upload/status/${dataResourceUid}"
             def start = System.currentTimeMillis()
             def maxTime = 60 * 60 * 1000 //2hr
+            task.message = "Uploading ..."
             while (start + maxTime > System.currentTimeMillis()) {
                 Thread.sleep(10000) // 10s
+                task.history.put(System.currentTimeMillis(), "checking status of " + statusUrl)
                 def txt = Util.getUrl(statusUrl)
                 if (txt == null) {
                     // retry
                 } else if (txt.contains("COMPLETE")) {
-                    task.message = "upload successful"
-
-                    //add species layer
-                    def species = [q   : "data_resource_uid:${dataResourceUid}",
-                                   ws  : sandboxHubUrl,
-                                   bs  : sandboxBiocacheServiceUrl,
-                                   name: name]
-
-                    addOutput("species", (species as JSON).toString())
-                    break
+                        task.history.put(System.currentTimeMillis(), "Uploading completed")
```

Comment:
Don't need to set `task.message` here?

Reply:
"Task completed" message is set by thread